Task 001

Starlight13 opened the pull request:
Refactored code to comform to MVVM + Provider pattern.

ascentman commented on lib/screens/shared_components/cart_button.dart:
```diff
@@ -0,0 +1,94 @@
+    ).animate(CurvedAnimation(parent: _controller, curve: Curves.bounceIn));
+  }
+
+  @override
```
try to organise your functions in that way:
- init
- build
- dispose
- private methods, started from underscore

ascentman reviewed (approved):
In general, good work! keep it up!💪

Starlight13 commented:
I have added localisation and refactored the code according to the comments.
I could not find any examples of viewModels communicating with each other in Flutter, and could not figure out how to apply methods used in other programming languages. So I came up with the following solution:
_To remove logic from product_details_screen I added my viewModels to sl, so that I can get them not only from context, but also from there. That way I can avoid passing context to viewModel and remove most of the logic from product_details_screen._ 
Is this solution suitable here?

ascentman commented:
> I have added localisation and refactored the code according to the comments.
> I could not find any examples of viewModels communicating with each other in Flutter, and could not figure out how to apply methods used in other programming languages. So I came up with the following solution:
> _To remove logic from product_details_screen I added my viewModels to sl, so that I can get them not only from context, but also from there. That way I can avoid passing context to viewModel and remove most of the logic from product_details_screen._
> Is this solution suitable here?

yes, sending viewModels to locator is one of the option and it fits well here. Good job!

ascentman commented on lib/screens/cart_screen/cart_screen.dart:
```diff
@@ -0,0 +1,214 @@
+                builder: (BuildContext context) {
+                  return AlertDialog(
+                    title: Text(
+                      AppLocalizations.of(context)!.clearCartTitle,
```
you can write something like that: `final localizations = CommonLocalizations.of(context);` below line 19:
`final viewModel = context.watch<ICartViewModel>();` and then use variable `localisations` in all cases. Make this change in all files
